From ebd48e4679f284a1c76441feef15c528867a718c Mon Sep 17 00:00:00 2001
From: Christian Schirge <schirge@zalando.de>
Date: Tue, 24 May 2022 11:05:39 +0200
Subject: [PATCH] float speed

the driver needs to return the speed as float instead of an integer
---
 napalm_arubaoss/helper/get_interfaces.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/napalm_arubaoss/helper/get_interfaces.py b/napalm_arubaoss/helper/get_interfaces.py
index 0949f1a..8c1674f 100644
--- a/napalm_arubaoss/helper/get_interfaces.py
+++ b/napalm_arubaoss/helper/get_interfaces.py
@@ -15,7 +15,7 @@ def get_interfaces(self):
         'is_enabled': False,
         'description': '',
         'last_flapped': -1.0,
-        'speed': 1000,
+        'speed': 1000.0,
         'mtu': -1,
         'mac_address': 'FA:16:3E:57:33:61',
     }
@@ -55,7 +55,7 @@ def get_interfaces(self):
         if i_id not in output.keys():
             output[i_id] = interface_template.copy()
 
-        output[i_id]["speed"] = speed
+        output[i_id]["speed"] = float(speed)
 
     for interface_id, interface_values in output.items():
         resp = self.connection.run_cmd(
